draco/core/containers.py

```diff
@@ -3340,3 +3340,227 @@ def copy_datasets_filter(
         if isinstance(dest_dset, memh5.MemDatasetDistributed):
             # Redistribute back to the original axis
             dest_dset.redistribute(original_ax_id)
+
+
+    # New container for the estimated bandpass gains and window
+class VisBandpassWindow(FreqContainer):
+    """Container for bandpass gains and their window estimated by running bandpass HyFoReS on hybrid beam-formed visibilities"""
+
+    _axes = ("pol", "freq") #Q8: do I need to declare _axes? What is the point of this? might have to add pol
+
+    _dataset_spec: ClassVar = {
+        "bandpass": {
+            "axes": ["pol", "freq"],
+            "dtype": np.complex128,
+            "initialise": True,
+            "distributed": False,
+        },
+        "window": {
+            "axes": ["pol", "freq", "freq"],
+            "dtype": np.complex128,
+            "initialise": True,
+            "distributed": False,
+        },
+    }
+
+    #_data_dset_name = "stack"
+    #_weight_dset_name = "weight"
+
+    @property
+    def bandpass(self):
+        """Get the bandpass dataset."""
+        return self.datasets["bandpass"]
+
+    @property
+    def window(self):
+        """Get the window dataset."""
+        return self.datasets["window"]
+
+class VisBandpassCompensate(FreqContainer):
+    """Container for window-compensated bandpass gains"""
+
+    _axes = ("pol", "freq")
+
+    _dataset_spec: ClassVar = {
+        "comp_bandpass": {
+            "axes": ["pol", "freq"],
+            "dtype": np.complex128,
+            "initialise": True,
+            "distributed": False,
+        },
+        "sval": {
+            "axes": ["pol", "freq"], # want an axis that has the same length as the freq axis "mode"
```

Comment:
What is the frequency axis here? If it's a series of modes with axis length equal to frequency, rather than the actual frequencies, you should create a new axis with a descriptive name